feature/checker-inbox-task/src/main/java/com/sparklead/feature/checker_inbox_task/checker_inbox_and_task/presentation/CheckerInboxTasksScreen.kt

```diff
@@ -0,0 +1,210 @@
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package com.sparklead.feature.checker_inbox_task.checker_inbox_and_task.presentation
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.ArrowBackIosNew
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import coil.compose.AsyncImage
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import com.mifos.core.designsystem.component.MifosCircularProgress
+import com.mifos.core.designsystem.component.MifosSweetError
+import com.mifos.core.designsystem.theme.Black
+import com.mifos.core.designsystem.theme.White
+import com.sparklead.feature.checker_inbox_task.R
+
+/**
+ * Created by Aditya Gupta on 21/03/24.
+ */
+
+@Composable
+fun CheckerInboxTasksScreen(onBackPressed: () -> Unit, checkerInbox: () -> Unit) {
+
+    val checkerInboxTasksViewModel: CheckerInboxTasksViewModel = hiltViewModel()
+    val state = checkerInboxTasksViewModel.checkerInboxTasksUiState.collectAsState().value
+    val isRefreshing by checkerInboxTasksViewModel.isRefreshing.collectAsState()
+    val swipeRefreshState = rememberSwipeRefreshState(isRefreshing = isRefreshing)
+
+    LaunchedEffect(key1 = true) {
+        checkerInboxTasksViewModel.loadCheckerTasksBadges()
+    }
+
+
+    Scaffold(
+        modifier = Modifier.fillMaxSize(),
+        containerColor = White,
+        topBar = {
+            TopAppBar(
```

Comment:
Move this in designsystem module and if we don't have generic class in designsystem them create one